libs/can/src/CANInterface.cpp

```diff
@@ -55,12 +55,8 @@ void CANInterface::rxPostman(void) {
 void CANInterface::rxClient(void) {
   while (true) {
     CANMsg *mail = nullptr;
-
-    // Wait for a message to arrive
-    do {
-      mail = m_rxMailbox.try_get();  // using try_get() because try_get_for() was crashing
-      ThisThread::sleep_for(1ms);
-    } while (mail == nullptr);
+    // Check if a message has arrived:
+    mail = m_rxMailbox.try_get(); 
 
     MBED_ASSERT(mail != nullptr);
```

Comment:
I agree with the above logic change on lines 58 and 59, but now it results in try_get sometimes returning nullptr if the mailbox is empty. should change this assert to maybe a debug print and early return(or just an if block to block execution of rest of this function. personally, i prefer single return statements from functions in fw code)

Reply:
Would the MBED_ASSERT not account for that?

Reply:
MBED_ASSERT would actually break the program, I agree that we should remove it and just wrap the logic in an if block